src/Components/Common/Hooks/sources.js

```diff
@@ -0,0 +1,70 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import { useQuery, useQueries } from 'react-query';
+
+import { SOURCES_QUERY_KEY, SOURCE_UPLOAD_INFO_KEY } from '../../../API/queryKeys';
+import { fetchSourcesList, fetchSourceUploadInfo } from '../../../API';
+import { AWS_PROVIDER, AZURE_PROVIDER } from '../../../constants';
+
+export const imageProps = PropTypes.shape({
+  name: PropTypes.string,
+  id: PropTypes.string,
+  provider: PropTypes.string,
+  architecture: PropTypes.string,
+  sourceIDs: PropTypes.arrayOf(PropTypes.string),
+  accountIDs: PropTypes.arrayOf(PropTypes.string),
+}).isRequired;
+
+export const useSourcesData = (provider, { refetch = false } = {}) => {
+  const {
+    error,
+    isLoading,
+    data: sources,
+  } = useQuery([SOURCES_QUERY_KEY, provider], () => fetchSourcesList(provider), {
+    enabled: !!provider,
+    refetchInterval: refetch && 10000,
+  });
+  return { error, isLoading, sources };
+};
+
+const useSourcesUploadInfos = (provider, { refetch }) => {
+  const { sources, isLoading: isLoadingSources, error: sourcesError } = useSourcesData(provider, { refetch });
+
+  const uploadInfos = useQueries(
+    sources?.map((source) => ({ queryKey: [SOURCE_UPLOAD_INFO_KEY, source.id], queryFn: () => fetchSourceUploadInfo(source.id), retry: 1 })) || []
+  );
+
+  const isLoading = isLoadingSources || uploadInfos.every((info) => info.isLoading);
+  const infos = [];
+  sources?.forEach((source, i) => {
+    uploadInfos[i].isSuccess && infos.push({ ...uploadInfos[i].data, ...source });
+  });
```

Comment:
no need to iterate if the queries are still loading, no?
```suggestion
  !isLoading && sources?.forEach((source, i) => {
    uploadInfos[i].isSuccess && infos.push({ ...uploadInfos[i].data, ...source });
  });
```